src/sqlfluff/cli/commands.py

```diff
@@ -902,8 +902,10 @@ def fix(
 
     if show_lint_violations:
         click.echo("==== lint for unfixable violations ====")
-        for violation in result.get_violations(**num_violations_kwargs):
-            click.echo(formatter.format_violation(violation))
+        for file, violations in result.violation_dict(**num_violations_kwargs).items():
```

Comment:
Does this potentially cause the files to be listed in a different order than before?

Reply:
Technically because of the way python's dictionary works past 3.6, insertion order is preserved when iterating over the item, but this code doesn't enforce that will stay this way. This logic iterates over `LintingResult.paths` in the same order as `LintingResults.get_violations` did though, which is my guess to be the same way that the per-file errors are ordered for fixable violations (though I haven't looked at that part of the code). 

If we want we can sort the possible files and then output them in order. No strong preference from my end as a consumer.

Reply:
Cool, let's enforce alphabetical order. If it's a lot of files, this could be helpful. Plus, getting consistent output on each run is nice as long as it's low or no cost (like here).